### PR TITLE
Add ProcessorFormatter for setting different renderers for different loggers

### PR DESCRIPTION
**Pull request by insolite**
…using native logging formatters.
Implementation taken from https://github.com/hynek/structlog/pull/73#issuecomment-222931463
Could be useful along with #78

### CODE REVIEW

**Comment by hynek** [comment]
(Just wanted to let you know that I’m currently too busy to review either of your PRs but am grateful for your submissions.  I’ll get to it eventually.  Sorry.)

**Comment by hynek** [comment]
Could you please give me an example on how this is supposed to be used?  I can write the docs myself if it needs to be but I’m a bit lost about its practical application?

**Comment by insolite** [comment]
For example, when we need to use colored output for console and plain text for file, we can configure logger like this:
```python
    logging.config.dictConfig({
        'version': 1,
        'disable_existing_loggers': False,
        'formatters': {
            'plain': {
                '()': ProcessorFormatter,
                'processor': structlog.dev.ConsoleRenderer(colors=False),
            },
            'colored': {
                '()': ProcessorFormatter,
                'processor': structlog.dev.ConsoleRenderer(colors=True),
            },
        },
        'handlers': {
            'default': {
                'level': 'DEBUG',
                'class': 'logging.StreamHandler',
                'formatter': 'colored',
            },
            'file': {
                'level': 'DEBUG',
                'class': 'logging.handlers.WatchedFileHandler',
                'filename': 'test.log',
                'formatter': 'plain',
            },
        },
        'loggers': {
            '': {
                'handlers': ['default', 'file'],
                'level': 'DEBUG',
                'propagate': True,
            },
        }
    })
    structlog.configure(
        processors=[
            structlog.stdlib.add_log_level,
            structlog.stdlib.PositionalArgumentsFormatter(),
            structlog.processors.TimeStamper(fmt='%Y-%m-%d %H:%M:%S'),
            structlog.processors.StackInfoRenderer(),
            structlog.processors.format_exc_info,
            event_dict_to_message,
        ],
        context_class=dict,
        logger_factory=structlog.stdlib.LoggerFactory(),
        wrapper_class=structlog.stdlib.BoundLogger,
        cache_logger_on_first_use=True,
    )
```

File: src/structlog/formatters.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        Process a copy of ``record.msg`` since the some processors modify the
+        ``event_dict`` and the ``LogRecord`` will be used for multiple
+        formatting runs.
+
+        """
+        if isinstance(record.msg, dict):
+            msg_repr = self.processor(
+                record._logger, record._name, record.msg.copy())
+            return msg_repr
+        return record.getMessage()
```
Comment (if-fi) [inline src/structlog/formatters.py]: I would suggest that you change the implementation to call the `super` formatter as well
``` python
        if isinstance(record.msg, dict):
            msg_repr = self.processor(
                record._logger, record._name, record.msg.copy())
            record.msg = msg_repr
        formatted_msg = super(ProcessorFormatter, self).format(record)
        return formatted_msg
```

that way we can configure additional formatting in the logging config e.g.
``` python
            'plain': {
                '()': ProcessorFormatter,
                'processor': structlog.dev.ConsoleRenderer(colors=False),
               'format': '\n%(message)s [in %(pathname)s:%(lineno)d]'
            },
```

**Comment by if-fi** [comment]
@insolite Thank you, for thins pull request!
This is a usecase that I have, and I have been trying to figure out how to make it work elegantly with `structlog`.
I have tried your code locally and it works perfectly.
I hope it will become part of the next version of `structlog`.

I have also added a small comment on the implementation.

File: src/structlog/formatters.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    """Custom stdlib logging formatter for structlog ``event_dict`` messages.
+
+    Apply a structlog processor to the ``event_dict`` passed as
+    ``LogRecord.msg`` to convert it to loggable format (a string).
+
+    """
+
+    def __init__(self, processor, *args, **kwargs):
+        """Keep reference to the ``processor``."""
+        super(ProcessorFormatter, self).__init__(*args, **kwargs)
```
Comment (if-fi) [inline src/structlog/formatters.py]: And we should have a default `'fromat'` if no `'format'` is set in the logging config.
``` python
        fmt = kwargs.pop('fmt', '%(message)s')
        super(ProcessorFormatter, self).__init__(fmt=fmt, *args, **kwargs)
```

**Comment by hynek** [comment]
c.f. #105 – sorry for the delays

**Pull request closed without merging**